Document operational requirements for Worker node. (#2279)
nucypher/nucypher: docs/source/guides/network_node/ursula_configuration_guide.rst

ether to pay for transaction gas; however, it is *not* necessary (and potentially risky) to hold NU tokens on a worker's
account for any reason.

Worker Requirements
-------------------

> 

Reviewer: Here are the operation requirement factors:

- Level 2 Availability, NAT management, Firewalls, etc.
- Peer metadata storage (grows with network size, each node is `2.0K`) 
- 2GB of free RAM to derive scrypt passwords
- 1GB RAM for normal runtime (needs profiling)
- CPU for re-encryptions (extremely light weight, see CI benchmarks)

Author: Made some updates to the list:
- I think breaking down the RAM requirements is a bit overkill no? I can change 4GB (minimum) to 3GB (minimum) if you rather.
- re: HDD storage, I think we should still put a sane minimum - any thoughts there?

Reviewer: Yeah - not meant as direct inclusions - just a way to think about the problem space and attempting to answer the question: *"What exactly dictates the requirements of nucypher?"*

Author: ✔️

Aside from the :ref:`base requirements <base-requirements>` for installation of the ``nucypher`` library:

* 4GB RAM (minimum)

> * 2 CPUs (minimum)

Reviewer: x86 architecture only (sorry, no raspberry pi 1, 2, or 3) -- IIRC RPI4+ have x86 versions?

Author: ✔️

> * 256GB HDD

Reviewer: This is too high IMO

Author: ✔️

* TCP Port 9151 opened for network communication
* Access to a fully synced Ethereum web3 provider e.g. local node, infura, alchemy etc. (see :ref:`using-eth-node`) is
  required to read and write to smart contracts

> 

Reviewer: ```suggestion
As a frame of reference - Maintaining an Ursula is similar to the deployment and maintenance of a high-availability web service, with the addition of ethereum accounts management.
```

In order to be a successful Ursula operator, you will need a machine (physical or virtual) which
can be kept online and ready to perform cryptographic operations for the network. However, short

> temporary service disruptions such as upgrades are understandable. Common cloud setups used
> include `AWS T3.medium <https://aws.amazon.com/ec2/instance-types/t3/>`_ and
> `Digital Ocean 4GB Basic Droplet <https://www.digitalocean.com/pricing/>`_.

Reviewer 2: ```suggestion
temporary service disruptions such as upgrades are understandable. Cloud setups are acceptable – for example,`Digital Ocean 4GB Basic Droplet <https://www.digitalocean.com/pricing/>`_ satisfies the memory and processing power requirements listed above.
```

We can't stop stakers using AWS (nor should we), but I reckon we should stop short at officially recommending the archetypal 'merchant of infrastructure' – this slogan is on our DAO page! 😇

Reviewer: If we're going to emphasize AWS :-1: Let's give some extra attention to what it means to run a node at home.  ( dappnode, desktops, lightweight servers, NUC, RPI, etc.)

.. important::s

    If also running a local Ethereum node on the same machine,
    `additional requirements <https://docs.ethhub.io/using-ethereum/running-an-ethereum-node/>`_ are needed.

Configure and Run a Worker
--------------------------

> It is assumed that you already have nucypher installed, have initiated a stake, and bonded a worker.

Reviewer: ```suggestion
 This guide assumes that you already have nucypher installed, have initiated a stake, and bonded a worker.
```

ether to pay for transaction gas; however, it is *not* necessary (and potentially risky) to hold NU tokens on a worker's
account for any reason.

Worker Requirements
-------------------

Aside from the :ref:`base requirements <base-requirements>` for installation of the ``nucypher`` library:

* 4GB RAM (minimum)
* 2 CPUs (minimum) - x86 architecture
* HDD storage requirement grows with network size - information on each node requires ~2KB of storage

> * Static public IP address - NAT management where applicable

Author: Is `static` fair, since it is allowed for the IP to change, but then the worker's configuration would need to be updated?

Reviewer: Hmmm, certainly much more desirable to have a static IP, but its not *technically* required.

Author: ✔️

Aside from the :ref:`base requirements <base-requirements>` for installation of the ``nucypher`` library:

* Dedicated physical/virtual machine
* Physical or SSH access
* 2GB RAM (minimum)
* x86 architecture
* 20GB HDD free storage - backups are required since data loss results in a malfunctioning worker
* Publicly available IP address - static where possible, NAT management where applicable
* TCP Port 9151 opened for network communication - firewall rules where applicable
* Access to a fully synced Ethereum web3 provider e.g. local node, Infura, Alchemy etc. (see :ref:`using-eth-node`) is
  required to read and write to smart contracts

..
    TODO: separate section on backups and data (#2285)

Workers can be run on cloud infrastructure – for example,
`Digital Ocean 4GB Basic Droplet <https://www.digitalocean.com/pricing/>`_ satisfies the memory and processing
power requirements listed above.

.. important::

    If also running a local Ethereum node on the same machine,
    `additional requirements <https://docs.ethhub.io/using-ethereum/running-an-ethereum-node/>`_ are needed.

Configure and Run a Worker
--------------------------

> This guide assumes that you already have nucypher installed, have initiated a stake, and bonded a worker.

Reviewer 3: ```suggestion
This guide assumes that you already have ``nucypher`` installed, have initiated a stake, and bonded a worker.
```

.. _fund-worker-account:

2. Fund Worker Account with ETH
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> Ensure that the worker's ethereum account has ETH for transaction gas.

Reviewer 2: ```suggestion
Ensure that the worker's ethereum account has sufficient ETH to pay the gas for regular transactions, or it may forgo subsidies (inflationary rewards). **Note that reducing the gas costs burdened upon stakers/workers is an active and high-priority area of network development.**
```
